tests/test_async_downloader.py

```diff
@@ -0,0 +1,149 @@
+import shutil
```

Comment:
instead of duplicating this file, could you make the original test_downloader test both the async and non async version ?